staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -41,6 +42,10 @@ const (
 	defaultRetries = 2
 	// protobuf mime type
 	mimePb = "application/com.github.proto-openapi.spec.v2@v1.0+protobuf"
+	// defaultTimeout is the maximum amount of time per request when no timeout has been set on a RESTClient.
+	// defaults to 32s in order to have a distinguishable length of time, relative to other timeouts that exist.
+	// we enforce a timeout for every request here in order to avoid hanging indefinitely
+	defaultTimeout = 12 * time.Second
```

Comment:
You're saying 32 in the comment but I see 12 here, typo?

Reply:
thanks for catching that, will update godoc - @deads2k I lowered the time to 12s as it is still distinguishable from other timeouts, long enough to avoid failing on "normal" levels of latency, yet still not as long such that if every discovery request times out, the overall time that a command hangs is reduced.

Reply:
> long enough to avoid failing on "normal" levels of latency

the openapi docs are quite large... on low-bandwidth connections, this seems too low

Reply:
+1 on leaving this at the previous 32.

Reply:
@soltysh @liggitt thanks, switched this back to 32s